js/braziliex.js

```diff
@@ -0,0 +1,425 @@
+"use strict";
+
+//  ---------------------------------------------------------------------------
+
+const Exchange = require ('./base/Exchange')
+const { ExchangeError, InsufficientFunds, OrderNotFound, InvalidOrder, AuthenticationError } = require ('./base/errors')
+
+//  ---------------------------------------------------------------------------
+
+module.exports = class braziliex extends Exchange {
+
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'id': 'braziliex',
+            'name': 'Braziliex',
+            'countries': 'BR',
+            'rateLimit': 1000,
+            // obsolete metainfo interface
+            'hasFetchTickers': true,
+            'hasFetchOpenOrders': true,
+            'hasFetchClosedOrders': true,
+            // new metainfo interface
+            'has': {
+                'fetchTickers': true,
+                'fetchOpenOrders': true,
+                'fetchClosedOrders': true,
+            },
+            'urls': {
+                'logo': 'https://braziliex.com/img/logo_topo.png',
+                'api': 'https://braziliex.com/api/v1',
+                'www': 'https://braziliex.com/',
+                'doc': 'https://braziliex.com/exchange/api.php',
+                'fees': 'https://braziliex.com/exchange/fees.php',
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'currencies',
+                        'ticker',
+                        'ticker/{market}',
+                        'orderbook/{market}',
+                        'tradehistory/{market}',
+                    ],
+                },
+                'private': {
+                    'post': [
+                        'balance',
+                        'complete_balance',
+                        'open_orders',
+                        'trade_history',
+                        'deposit_address',
+                        'sell',
+                        'buy',
+                        'cancel_order',
+                    ],
+                },
+            },
+            'fees': {
+                'trading': {
+                    'maker': 0.005,
+                    'taker': 0.005,
+                },
+            },
+            'precision': {
+                'amount': 8,
+                'price': 8,
+            },
+        });
+    }
+
+    async fetchCurrencies (params = {}) {
+        let currencies = await this.publicGetCurrencies (params);
+        let ids = Object.keys (currencies);
+        let result = {};
+        for (let i = 0; i < ids.length; i++) {
+            let id = ids[i];
+            let currency = currencies[id];
+            let precision = currency['decimal'];
+            let code = this.commonCurrencyCode (id.toUpperCase ());
```

Comment:
This one is the same as in fetchMarkets, if you can split it in two lines... but if you don't want to, don't bother, I will do that upon merging.